src/ixdat/measurements.py

```diff
@@ -1,35 +1,73 @@
-An ixdat Dataset is a collection of references to DataSeries with the metadata required
+An ixdat Measurement is a collection of references to DataSeries with the metadata needed
 to combine them, i.e. "build" the combined dataset. It has a number of general methods
-to visualize and analyze the combined dataset. Dataset is also the base class for a
-number of technique-specific Dataset-derived classes.
+to visualize and analyze the combined dataset. Measurement is also the base class for a
+number of technique-specific Measurement-derived classes.
+
+A Measurement will typically be accompanied by one or more Calibration. This module
+also defines the base class for Calibration, while technique-specific Calibration
+classes will be defined in the corresponding module in ./techniques/
 """
 import json
 import numpy as np
-from .db import Saveable, PlaceHolderObject
-from .data_series import DataSeries, TimeSeries, ValueSeries
+from .db import Saveable, PlaceHolderObject, fill_object_list
+from .data_series import (
+    DataSeries,
+    TimeSeries,
+    ValueSeries,
+    ConstantValue,
+    append_series,
+    time_shifted,
+)
 from .samples import Sample
 from .lablogs import LabLog
-from .exporters import CSVExporter
-from .exceptions import BuildError, TechniqueError, SeriesNotFoundError
+from .exporters.csv_exporter import CSVExporter
+from .plotters.value_plotter import ValuePlotter
+from .exceptions import BuildError, SeriesNotFoundError
 
 
 class Measurement(Saveable):
     """The Measurement class"""
 
+    # ------ table description class attributes --------
     table_name = "measurement"
     column_attrs = {
         "name",
         "technique",
         "metadata",
+        "aliases",
         "sample_name",
         "tstamp",
     }
     extra_linkers = {
-        "measurement_series": ("data_series", "s_ids"),
         "component_measurements": ("measurements", "m_ids"),
+        "measurement_calibrations": ("calibration", "c_ids"),
+        "measurement_series": ("data_series", "s_ids"),
+    }
+    child_attrs = ["component_measurements", "calibration_list", "series_list"]
+    # TODO: child_attrs should be derivable from extra_linkers?
+
+    # ---- measurement class attributes, can be overwritten in inheriting classes ---- #
+    control_technique = None
+    """Name of technique primarily used to control the experiment"""
+    control_str = None
+    """Name (or alias) for main time variable or main time-dependent value variable,
+    typically of the control_technique"""
+    sel_str = "selector"
+    """Name of the default selector"""
+    select_on = ("file_number",)
+    """Name of the default things to use to construct the selector"""
+    series_constructors = {
+        "file_number": "_build_file_number",
+        "selector": "_build_selector",
     }
+    """Series which should be constructed from other series by the specified method
+    and cached the first time they are looked up"""
+    essential_series = None
```

Comment:
NAME: I think these are also names right, maybe the same applies as above.

Reply:
done.